arch/xtensa/src/esp32s3/esp32s3_gpio.c

```diff
@@ -141,25 +202,37 @@ int esp32s3_configgpio(int pin, gpio_pinattr_t attr)
  * Name: esp32s3_gpio_matrix_in
  *
  * Description:
- *   Set gpio input to a signal
- *   NOTE: one gpio can input to several signals
- *   If gpio == 0x3c, cancel input to the signal, input 0 to signal.
- *   If gpio == 0x3a, input nothing to signal.
- *   If gpio == 0x38, cancel input to the signal, input 1 to signal.
+ *   Set GPIO input to a signal.
+ *   NOTE: one GPIO can input to several signals.
+ *
+ * Input Parameters:
+ *   pin           - GPIO pin to be configured.
+ *                   - If pin == 0x3c, cancel input to the signal, input 0
+ *                     to signal.
+ *                   - If pin == 0x3a, input nothing to signal.
+ *                   - If pin == 0x38, cancel input to the signal, input 1
+ *                     to signal.
+ *   signal_idx    - Signal index.
+ *   inv           - Flag indicating whether the signal is inverted.
+ *
+ * Returned Value:
+ *   None.
  *
  ****************************************************************************/
 
-void esp32s3_gpio_matrix_in(uint32_t gpio, uint32_t signal_idx, bool inv)
+void esp32s3_gpio_matrix_in(uint32_t pin, uint32_t signal_idx, bool inv)
 {
   uint32_t regaddr = GPIO_FUNC0_IN_SEL_CFG_REG + (signal_idx * 4);
```

Comment:
Sorry for dummy question, but maybe??
```suggestion
  uint32_t regaddr = GPIO_FUNC0_IN_SEL_CFG_REG + (pin * 4);
```
I'm just looking into `esp32s3_gpio_matrix_out` implementation so this question rises.

Reply:
Indeed, it seems catchy, but here the offset is based on `signal_idx`.
Here the driver is configuring the GPIO Matrix to forward the input value from a pin (0-48) to a peripheral signal (0-256). So, for each of the 256 peripheral signals, there is a `GPIO_FUNCx_IN_SEL_CFG_REG` register.

Reply:
Ok. NP. I just want to make sure that all is fine, so double checking the details.